src/vring.rs

```diff
@@ -94,38 +94,42 @@ pub trait VringT<M: GuestAddressSpace>:
 /// This struct maintains all information of a virito queue, and could be used as an `VringT`
 /// object for single-threaded context.
 pub struct VringState<M: GuestAddressSpace = GuestMemoryAtomic<GuestMemoryMmap>> {
-    queue: Queue<M>,
+    queue: Queue,
     kick: Option<EventFd>,
     call: Option<EventFd>,
     err: Option<EventFd>,
     enabled: bool,
+    mem: M,
 }
 
 impl<M: GuestAddressSpace> VringState<M> {
     /// Create a new instance of Vring.
     fn new(mem: M, max_queue_size: u16) -> Self {
         VringState {
-            queue: Queue::new(mem, max_queue_size),
+            // TODO: We probably need to propagate this error instead.
```

Comment:
I agree. The error needs to be propagated and made visible to the users of the crate through `VringMutex` and `VringRwLock`.

Reply:
Done.